### PR TITLE
Object selection: Part 2

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Based on #2128. This implements the functions that prepare the object selection.

### CODE REVIEW

**Review by LeftofZen — COMMENTED** [review]
couple files have the same changes as https://github.com/OpenLoco/OpenLoco/pull/2128 which I left comments on, so if those changes are implemented/merge first then this PR should be ok. If this PR is merged first, the comments from that PR should be applied here

File: src/OpenLoco/src/Objects/ObjectManager.h [hunk 1]
```diff
@@ -103,8 +103,7 @@ namespace OpenLoco::ObjectManager
         return counts[(size_t)type];
     };
 
-    constexpr size_t maxObjects = 859;
-    constexpr size_t maxObjectTypes = 34;
+    constexpr size_t kMaxObjects = 859;
```
Comment (LeftofZen) [inline src/OpenLoco/src/Objects/ObjectManager.h:106]: Should be in S5/Limits.h

File: src/OpenLoco/src/Objects/Object.h [hunk 2]
```diff
@@ -45,6 +45,7 @@ namespace OpenLoco
         scenarioText,
     };
 
+    constexpr size_t kMaxObjectTypes = 34;
```
Comment (LeftofZen) [inline src/OpenLoco/src/Objects/Object.h:48]: S5/Limits.h

File: src/OpenLoco/src/Objects/ObjectIndex.cpp [hunk 3]
```diff
@@ -472,6 +507,238 @@ namespace OpenLoco::ObjectManager
...
+        // With also load objects it doesn't matter if they can't load
+        for (const auto& header : entry._alsoLoadObjects)
+        {
+            if ((mode & SelectObjectModes::selectAlsoLoads) != SelectObjectModes::none)
+            {
+                selectObjectFromIndexInternal(mode, true, header, objectFlags, selectionMetaData);
+            }
+        }
+
+        // When in don't load depenedent objects (required or also loads) mode then
```
Comment (LeftofZen) [inline src/OpenLoco/src/Objects/ObjectIndex.cpp:616]: typo + grammar

File: src/OpenLoco/src/Objects/ObjectIndex.cpp [hunk 4]
```diff
@@ -510,4 +779,260 @@ namespace OpenLoco::ObjectManager
 
         return entry;
     }
+
+    // 0x00472D3F
+    static void markInUseObjects([[maybe_unused]] std::span<SelectedObjectsFlags> objectFlags)
```
Comment (LeftofZen) [inline src/OpenLoco/src/Objects/ObjectIndex.cpp:784]: this method is quite large - is it possible to break into smaller pieces?

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Objects/ObjectIndex.cpp:779]: Seconding this remark.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**